### PR TITLE
fix: Check proxy protocols explicitly

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: electron/src/main.ts [hunk 1]
```diff
@@ -104,6 +104,9 @@ if (argv.version) {
 if (argv[ARG.PROXY_ARG]) {
   try {
     proxyInfoArg = new URL(argv[ARG.PROXY_ARG]);
+    if (proxyInfoArg.protocol !== 'http:' && proxyInfoArg.protocol !== 'https:') {
```
Comment (ffflorian) [inline electron/src/main.ts]: What about `socks5://`? :)

Comment (Yserz) [reply electron/src/main.ts]: Done

File: electron/src/main.ts [hunk 2]
```diff
@@ -104,6 +104,14 @@ if (argv.version) {
 if (argv[ARG.PROXY_ARG]) {
   try {
     proxyInfoArg = new URL(argv[ARG.PROXY_ARG]);
+    if (
+      proxyInfoArg.protocol !== 'http:' &&
```
Comment (bennyn) [inline electron/src/main.ts]: Can this be done: `if (!['http:', 'https:', 'socks4:', 'socks5:'].includes(proxyInfoArg.protocol))`?

Comment (ffflorian) [reply electron/src/main.ts]: Or:
```js
if (!/^(https?|socks[45]):/.test(proxyInfoArg.protocol))
```

**Pull request merged**